spec/haskell/src/SEL4/Object/Structures.lhs

```diff
@@ -492,3 +496,15 @@ Various operations on the free index of an Untyped cap.
 > data TcbQueue = TcbQueue {
 >     tcbQueueHead :: Maybe (PPtr TCB),
 >     tcbQueueEnd :: Maybe (PPtr TCB) }
+
+\subsubsection{TCB Flags}
+
+> type TcbFlags = Word
+
+> data TcbFlag
+>     = NoFlag
```

Comment:
I'm still not sure I understand the purpose of `NoFlag`, could you enlighten me?

Reply:
I can't speak about the original intentions in the C, but in my mind here it provides a way to abstract away the details of how the flags are implemented. For instance, we could say `flags = NoFlag` to say that no flags are set.

Having now said that, this is another argument for adding something like the `flagSet` helper function I mentioned in a different comment, since that would extend this abstraction and hide the details of the flags being implemented as a `Word`.